### DISCUSSION TITLE
How do I create an homogeneous graph like an heterogeneous graph?

### DISCUSSION THREAD

**Discussion by ppsmk388** [pyg-team/pytorch_geometric#5977] [Q&A] [answered]
It is very easy to create nodes as well as edges when create the heterogeneous graph. 
https://pytorch-geometric.readthedocs.io/en/latest/notes/heterogeneous.html#creating-heterogeneous-graphs
![image](https://user-images.githubusercontent.com/60417397/201744752-3aff223a-e86f-47a2-a3ed-8e326df099c8.png)

Is there a similar way to create homogeneous graph like create the heterogeneous graph?
I don't see a similar method in the documentation
Thanks!

**Comment by EdisonLeeeee** [comment, marked as answer]
A homogeneous graph in PyG is denoted by `torch_geometric.data.Data`
```python
from torch_geometric.data import Data
data = Data(x=..., edge_index=..., edge_attr=...)
```

**Reply by ppsmk388** [reply]
Thank you!

**Reply by ppsmk388** [reply]
And can I find after the 
homogeneous_data = data.to_homogeneous()
After the above operation, The y assigned in the heterogeneous Graph disappears
How can I keep this y?

**Reply by EdisonLeeeee** [reply]
There should exist a `y` after `data.to_homogeneous()` if it already existed in the original heterogeneous graph. Could you specify?

**Reply by ppsmk388** [reply]
![image](https://user-images.githubusercontent.com/60417397/202881260-b99bb7b6-38bc-4cec-a800-faba95412d78.png)

![image](https://user-images.githubusercontent.com/60417397/202881217-9e42697b-bff9-4990-b300-a55e936e64e2.png)

Thank you for your reply!

In this example the label of dataset disappear when use the to_homogeneous() .


https://pytorch-geometric.readthedocs.io/en/latest/notes/heterogeneous.html#utility-functions

**Reply by EdisonLeeeee** [reply]
I've run the code and got the results:
```python
homogeneous_data = data.to_homogeneous()
print(homogeneous_data)

Data(edge_index=[2, 13605929], x=[1879778, 128], year=[1879778], y=[1879778], train_mask=[1879778], val_mask=[1879778], test_mask=[1879778], node_type=[1879778], edge_type=[13605929])
```

The goal of this tutorial is to give an illustration on some utility functions of [torch_geometric.data.HeteroData](https://pytorch-geometric.readthedocs.io/en/latest/modules/data.html#torch_geometric.data.HeteroData) class, and it only displays some important details for the sake of clarity. Sorry for the confusion, do you think we should update the tutorial?

**Reply by rusty1s** [reply]
Yes, we would need to update the tutorial since support for this just got recently added.

**Reply by ppsmk388** [reply]
Thanyk you!
